fuels-contract/src/contract.rs

```diff
@@ -82,16 +82,23 @@ impl Contract {
         wallet: LocalWallet,
     ) -> Result<Vec<Receipt>, Error> {
         // Script to call the contract. The offset that points to the `script_data` is loaded at the
-        // register `0x12`. Note that we're picking `0x12` simply because it could be any
+        // register `0x13`. Note that we're picking `0x13` simply because it could be any
```

Comment:
why did 0x12 changed to 0x13 if "any non-reserved" register can be picked?

Reply:
As the address are not reserved. We have chosen to keep an order for adding forwarding values first and then the data offset.

For now, looks like the order will be;
- 0x10 call data offset
- 0x11 gas_price
- 0x12 amount
- 0x13 asset_id

These are soft rules as addresses can be changed

Reply:
These soft rules make sense! But it shoud probably be written in the docstring imo